fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -455,8 +455,12 @@ class ReadIteratorPool {
 	void update() {
 		if (SERVER_KNOBS->ROCKSDB_READ_RANGE_REUSE_ITERATORS ||
 		    SERVER_KNOBS->ROCKSDB_READ_RANGE_REUSE_BOUNDED_ITERATORS) {
-			std::lock_guard<std::mutex> lock(mutex);
-			iteratorsMap.clear();
+			mutex.lock();
+			// The latest index might contain the current iterator which is getting created.
+			// But, that should be ok to avoid adding more code complexity.
+			deletedUptoIndex = index;
+			mutex.unlock();
+			deleteIteratorsPromise.send(Void());
```

Comment:
In flow, the `send` here means we synchronously invoke the callback, i.e., `readIterPool->refreshIterators();` in the `refreshReadIteratorPool()`, so this doesn't seem to solve the problem.

Reply:
The solution is probably 
1) remove this line and `waitNext(readIterPool->getDeleteIteratorsFuture())` in `refreshReadIteratorPool()` so that we just periodically delete iterators; or
2) wrap this line in an actor that can be scheduled asynchronously.

Reply:
@jzhou77 I totally assumed its a asynchronous call :(
 Periodical deletion is happening every 5 seconds.. 5 seconds might be too long as the read iterators hold some resources until it is freed. It might block the writes getting flushed and also compaction.
I added a wait in refreshReadIteratorPool() before calling refreshIterators.  Do you see any issues with this?

Reply:
Is it still a synchronous call?

+1 for the following approach
>2. wrap this line in an actor that can be scheduled asynchronously.